### PR TITLE
Rebase rustc-rayon on rayon-1.2

### PR DESCRIPTION
**Pull request by cuviper**
See also https://github.com/rust-lang/rustc-rayon/pull/3

### CODE REVIEW

**Comment by cuviper** [comment]
Let's try full CI to exercise parallel queries...

@bors try

**Comment by Mark-Simulacrum** [comment]
Ah, bors try is not going to run full CI at this point (and we don't actually have a parallel test builder, only a check builder for --cfg parallel)

**Comment by cuviper** [comment]
> Ah, bors try is not going to run full CI at this point

Just because of the merge conflict? I rebased again for `Cargo.lock` now.

> (and we don't actually have a parallel test builder, only a check builder for --cfg parallel)

How have other parallel PRs been tested? Should I flip it on somewhere just to try it out?

**Comment by cuviper** [comment]
Tidy doesn't like my (temporary) git source replacement, which is understandable.

File: Cargo.lock [hunk 1]
```diff
@@ -1911,12 +1887,6 @@ dependencies = [
  "winapi 0.3.6",
 ]
 
-[[package]]
-name = "memoffset"
-version = "0.2.1"
```
[the remark this answers is not in the document]
Comment (RalfJung) [reply Cargo.lock]: Yes, this is great, thanks :)

**Comment by Zoxc** [comment]
@cuviper I usually just test the PRs locally with `[rust] parallel-compiler = true` in `config.toml`.

**Comment by Zoxc** [comment]
@cuviper That's a preexisting failure. (and so is the thread names, but fixing those doesn't hurt)

**Comment by alexcrichton** [comment]
r? @cuviper 

(feel free to reassign back to me or someone else once this is ready to go)

**Comment by hdhoang** [comment]
ping from triage @cuviper, how is progress on this PR?

**Comment by cuviper** [comment]
Waiting on https://github.com/rust-lang/rustc-rayon/pull/3 -- I'll ping there.

**Comment by JohnCSimon** [comment]
Ping from triage.
@cuviper This PR has sat idle for the last 8 days - I see this as blocked from rust-lang/rustc-rayon#3 
Can you address this?

Thanks.

**Comment by cuviper** [comment]
r? @nikomatsakis 

The old stuff is still referenced by `rustc-ap-rustc_data_structures`, but I think that may be resolved once the real `rustc_data_structures` lands here, so a new one is published...

**Review by nikomatsakis — APPROVED** [review]

**Comment by nikomatsakis** [comment]
r=me

**Comment by RalfJung** [comment]
> The old stuff is still referenced by rustc-ap-rustc_data_structures, but I think that may be resolved once the real rustc_data_structures lands here, so a new one is published...

So what does it take to bump these? They currently keep a lot of old crates in our dependency tree:
```
[[package]]
name = "rustc-ap-rustc_data_structures"
version = "606.0.0"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "ca545744a5a9b42e3d0410d6290d40de96dd567253fe77f310c1de4afd213dd4"
dependencies = [
 "cfg-if",
 "crossbeam-utils 0.6.5",
 "ena",
 "indexmap",
 "jobserver",
 "lazy_static 1.3.0",
 "log",
 "parking_lot 0.9.0",
 "rustc-ap-graphviz",
 "rustc-ap-rustc_index",
 "rustc-ap-serialize",
 "rustc-hash",
 "rustc-rayon 0.2.0",
 "rustc-rayon-core 0.2.0",
 "smallvec",
 "stable_deref_trait",
]
```

**Comment by RalfJung** [comment]
Ah, looks like [`rustfmt` needs a bump](https://github.com/rust-lang/rustfmt/blob/master/Cargo.toml).

**Comment by mati865** [comment]
> Ah, looks like rustfmt needs a bump .

It's more complicated.
Only single version of `rustc-ap-*` can exist in Cargo.lock and it's pulled by `racer` (RLS dependency) and Rustfmt.
In the end it has to be coordinated among 3 different repositories.

**Pull request merged**